dev_sandbox/main.cpp

```diff
@@ -0,0 +1,672 @@
+
+#include "map/framework.hpp"
+
+#include "platform/platform.hpp"
+#include "platform/settings.hpp"
+
+#include "coding/reader.hpp"
+
+#include "base/logging.hpp"
+#include "base/macros.hpp"
+
+#include "std/target_os.hpp"
+
+#include <chrono>
+#include <functional>
+#include <mutex>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <gflags/gflags.h>
+
+#if defined(OMIM_OS_WINDOWS)
+#define GLFW_EXPOSE_NATIVE_WIN32
+#elif defined(OMIM_OS_LINUX)
+#define GLFW_EXPOSE_NATIVE_X11
+#elif defined(OMIM_OS_MAC)
+#define GLFW_EXPOSE_NATIVE_COCOA
+#else
+#error Unsupported plaform
+#endif
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h>
+#include <GLFW/glfw3native.h>
+#include <imgui/backends/imgui_impl_glfw.h>
+#include <imgui/imgui.h>
+
+DEFINE_string(data_path, "", "Path to data directory.");
+DEFINE_string(log_abort_level, base::ToString(base::GetDefaultLogAbortLevel()),
+              "Log messages severity that causes termination.");
+DEFINE_string(resources_path, "", "Path to resources directory.");
+DEFINE_string(lang, "", "Device language.");
+
+#if defined(OMIM_OS_MAC) || defined(OMIM_OS_LINUX)
+drape_ptr<dp::GraphicsContextFactory> CreateContextFactory(GLFWwindow * window, dp::ApiVersion api, m2::PointU size);
+void PrepareDestroyContextFactory(ref_ptr<dp::GraphicsContextFactory> contextFactory);
+void OnCreateDrapeEngine(GLFWwindow * window, dp::ApiVersion api, ref_ptr<dp::GraphicsContextFactory> contextFactory);
+void UpdateContentScale(GLFWwindow * window, float scale);
+void UpdateSize(ref_ptr<dp::GraphicsContextFactory> contextFactory, int w, int h);
+#endif
+
+#if defined(OMIM_OS_LINUX)
+// Workaround for storage::Status compilation issue:
+// /usr/include/X11/Xlib.h:83:16: note: expanded from macro 'Status'
+#undef Status
+#endif
+
+namespace
+{
+bool ValidateLogAbortLevel(char const * flagname, std::string const & value)
+{
+  if (auto level = base::FromString(value); !level)
+  {
+    std::cerr << "Invalid value for --" << flagname << ": " << value << ", must be one of: ";
+    auto const & names = base::GetLogLevelNames();
+    for (size_t i = 0; i < names.size(); ++i)
+    {
+      if (i != 0)
+        std::cerr << ", ";
+      std::cerr << names[i];
+    }
+    std::cerr << '\n';
+    return false;
+  }
+  return true;
+}
+
+bool const g_logAbortLevelDummy = gflags::RegisterFlagValidator(&FLAGS_log_abort_level, &ValidateLogAbortLevel);
+
+void errorCallback(int error, char const * description) { LOG(LERROR, ("GLFW (", error, "):", description)); }
+
+std::function<void(int w, int h)> onResize;
+std::function<void(double x, double y, int button, int action, int mods)> onMouseButton;
+std::function<void(double x, double y)> onMouseMove;
+std::function<void(double x, double y, double xOffset, double yOffset)> onScroll;
+std::function<void(int key, int scancode, int action, int mods)> onKeyboardButton;
+std::function<void(float xscale, float yscale)> onContentScale;
```

Comment:
nit: I'd prefer to make one Window or App instance here that holds all these callbacks. To have one global object instead of many.